### PR TITLE
Replace require with import homepage

### LINKED ISSUE #2465: Replace `require` with `import` for images on the homepage
## Problem

We use `require` for images on the homepage. `require` will be deprecated in Nuxt 3.

## Description

We should move the `hompage_images` folder (images only, not the `json` file) to `/static` folder, and use the static URLs.

https://github.com/WordPress/openverse/blob/507d7401a774aad70b984ae7237183d510b0de30/frontend/src/components/VHomeGallery/VHomeGallery.vue#L115-L117

Change the code above to 
```
src: `/homepage_images/${imageSet.value.key}/${idx + 1}.png`
```

### PR DESCRIPTION
**Pull request by sehgxl**
## Fixes

Fixes #2465 by @obulat 

## Description
Require is deprecated in nuxt3. So moved the images to static folder and used them from there.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — CHANGES_REQUESTED** [review]
This works perfectly, @sehgxl!

While this wasn't in the issue description, but could you also rename the `image_info.json` file to `homepage_images.json` and move it out of the `assets/homepage_images` folder into the `assets` folder? So, the import would become `import imageInfo from "~/assets/homepage_images.json"`.

**Comment by sehgxl** [comment]
@obulat I have pushed the changes, But there is one thing I wanted to ask about.

Since changes required were quite small, so I was able to navigate easily but I am having some errors while trying to run things locally. I am unable to resolve them, so please take a look.

Screenshot from 2023-07-11 17-39-09

In the terminal, I see that `client` gets stuck at 95%.

Screenshot from 2023-07-11 17-39-34

File: frontend/src/components/VHomeGallery/VHomeGallery.vue [hunk 1]
```diff
@@ -55,7 +55,7 @@ import useResizeObserver from "~/composables/use-resize-observer"
 
 import VLink from "~/components/VLink.vue"
 
-import imageInfo from "~/assets/homepage_images/image_info.json"
+import imageInfo from "~/assets/ homepage_images.json"
```
Comment (obulat) [inline frontend/src/components/VHomeGallery/VHomeGallery.vue:58]:
```suggestion
import imageInfo from "~/assets/homepage_images.json"
```
There is an extra space here. The file name also has a space, could you please remove it?

**Comment by obulat** [comment]
> I am having some errors while trying to run things locally. I am unable to resolve them, so please take a look.

How did you install dependencies, @sehgxl? Could you try to run `just install` (or `just node-install` if you don't want to deal with python dependencies) one more time, and then re-run `just frontend/run dev`?

File: frontend/src/components/VHomeGallery/VHomeGallery.vue (hunk 1 above)
Comment (sehgxl) [reply frontend/src/components/VHomeGallery/VHomeGallery.vue]: Pushed the changes, aplogies for the silly mistake.

**Comment by obulat** [comment]
@sehgxl, the last commit deleting the lock file should be reverted.

**Comment by dhruvkb** [comment]
@sehgxl I faced the same error yesterday. You might want to try this step to compile the Openverse ESLint plugin.

```
$ pnpm --filter='@openverse/eslint-plugin' build
```

Restarting the dev server should work after this^.

**Comment by sehgxl** [comment]
> @sehgxl I faced the same error yesterday. You might want to try this step to compile the Openverse ESLint plugin.
> 
> ```
> $ pnpm --filter='@openverse/eslint-plugin' build
> ```
> 
> Restarting the dev server should work after this^.

Thanks @dhruvkb, its working now.

File: frontend/src/components/VHomeGallery/VHomeGallery.vue (hunk 1 above)
Comment (obulat) [reply frontend/src/components/VHomeGallery/VHomeGallery.vue]: No problem, it happens to everyone :)

**Comment by obulat** [comment]
> Thanks dhruvkb, its working now.

Glad you've found the fix :) If you revert the changes to `pnpm-lock.yaml` file, @sehgxl, this PR would be good to go 🚀

**Comment by sehgxl** [comment]
@obulat Checkout the last commit, I have reverted the `pnpm-lock.yaml `. But for some reason a test is failing.

**Comment by obulat** [comment]
> @obulat Checkout the last commit, I have reverted the `pnpm-lock.yaml`. But for some reason a test is failing.

There were still some changes to `pnpm-lock` present in this PR. I rebased your PR onto `main`, and this removed those changes.

**Review by obulat — APPROVED** [review]

**Comment by obulat** [comment]
Thank you for your contribution, @sehgxl ✨

**Review by dhruvkb — APPROVED** [review]
LGTM, initially I was worried when I saw 47 files changed 😆. Thanks for the contribution @sehgxl!

**Comment by sehgxl** [comment]
Thanks for your help and patience @obulat and @dhruvkb, this is my first open source contribution that seems to be merging successfully. Glad that I was able to provide some value while learning along the way :)

**Comment by obulat** [comment]
> Thanks for your help and patience @obulat and @dhruvkb, this is my first open source contribution that seems to be merging successfully. Glad that I was able to provide some value while learning along the way :)

Congratulations on your first contribution, @sehgxl! Feel free to look at other issues :)

**Pull request merged**